tests/test_client_regress/tests.py

```diff
@@ -1189,11 +1189,11 @@ def test_patch(self):
     def test_empty_string_data(self):
         "Request a view with empty string data via request method GET/POST/HEAD"
         # Regression test for #21740
-        response = self.client.get('/body/', data='', content_type='application/json')
+        response = self.client.get('/body/', data='', content_type='text/plain')
```

Comment:
What's the rationale for this change? See https://code.djangoproject.com/ticket/21740#comment:6 for the original regression.

Reply:
So the rationale here is that...

 - That given example does not appear to rely on being JSON specifically – it's just an example.
 - The expected serialisation for `''` with a JSON content type is `b'""'`, not `b''`.
 - We can't drop the `content_type` parameter here because it defaults to multipart data, which won't serialise a string.

This is a behaviour change, but one that makes the behaviour more uniform given the expectation that JSON content is serialised automatically.

Reply:
My impression is that the example from the comment was actual code from the commenter's project. I'm not sure if a string should be encoded again as this would be backwards incompatible (I think) if someone is already encoding their data pre-Django 2.1 (i.e. using the advice in the documentation "Use json.dumps() on data before passing it to post()" (I didn't test it, so let me know if that's a wrong assumption).

Reply:
Hmmm. Good point, that isn't backwards incompatible in that case. Do you think it's best to special case strings in that case?

To do that, I would:

 - Add a special case
 - Remove the string test
 - Update the docs
 - Revert the regression test change
 - Add another regression test with an explicit example of a json dumped string.

Reply:
Alternatively, we could leave this one entirely and create a new optional `json` parameter to these functions, that sets the content type by default, and encodes everything passed to it as JSON.

Thoughts?

Reply:
I've gone ahead and implemented the first option here, as the second would probably be a bigger discussion. Let me know if you want me to do the latter though – I'd do it in a separate PR, but I'd be happy to do it.